src/importNameRule.ts

```diff
@@ -183,7 +197,11 @@ function walk(ctx: Lint.WalkContext<Option>) {
         moduleName: string,
         node: ts.ImportEqualsDeclaration | ts.ImportDeclaration
     ): boolean {
-        if (expectedImportedName === importedName) {
+        if (option.config.case === StringCase.any) {
```

Comment:
This is las one from my side.
With `transformName` + `indexOf` (yeah, I forgot about Node 6) there is no need for special handling of `StringCase.any`.
`transformName(expectedImportedName).indexOf(importedName) > -1` should catch all cases now.